include/util/deallocating_vector.hpp

```diff
@@ -239,6 +239,19 @@ class DeallocatingVector
         bucket_list.emplace_back(new ElementT[ELEMENTS_PER_BLOCK]);
     }
 
+    DeallocatingVector(DeallocatingVector &&other)
+    {
+        bucket_list = std::move(other.bucket_list);
+        current_size = std::move(other.current_size);
+    }
+
+    DeallocatingVector &operator=(DeallocatingVector &&other)
+    {
+        bucket_list = std::move(other.bucket_list);
+        current_size = std::move(other.current_size);
+        return *this;
+    }
```

Comment:
1/ Check the move ctor / assign op [here](https://github.com/Project-OSRM/osrm-backend/blob/f650aa9df2cee6e99c994e2964d4c84c0a589e04/include/util/deallocating_vector.hpp#L246-L257): aren't we missing to copy over the `current_size` member?

2/ what I meant was:

    DeallocatingVector(DeallocatingVector &&) = default;
    DeallocatingVector &operator=(DeallocatingVector &&) = default;

Isn't this what we are trying to do manually here?